### PR TITLE
Cages GA attestation 

### PR DESCRIPTION
**Pull request by hanneary**
# Why
We have revamped how attestation works for Cages before we make them GA and to use trusted certs instead of self signed

# How
- Get the attestation doc from `https://${cageName}.${appUuid}.${config.cagesHostname}/.well-known/attestation` and cache
- Poll periodically to get a fresh attestation doc before 3 hour expiry (currently 7200 seconds - 2 hours)
- Use rust bindings to verify the tls cert from the cage corresponds with the attestation doc
- If attestation fails, refresh cache and try again in case deployment has caused PCRs to change
- Existing(deprecated) attestation is now referred to as beta and new flow is GA

# Checklist

- [ ] Commit messages are formatted as per [the convention](https://www.conventionalcommits.org/en/v1.0.0/), and breaking changes are marked

### CODE REVIEW

File: env
Comment (donaltuohy) [inline env:1]: Is this and the export file below meant to be included?

**Review by donaltuohy — APPROVED** [review]
Nice!

File: lib/core/cageAttestationDoc.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    let cache = {};
+    await Promise.all(
+      this.cages.map(async (cageName) => {
+        const response = await this.http.getCageAttestationDoc(
+          cageName,
+          this.appUuid
+        );
+        cache[cageName] = response.attestation_doc;
+      })
+    );
```
Comment (lfarrel6) [inline lib/core/cageAttestationDoc.js:65]: we need to handle the request failing in the poll function. Not sure what the expected behaviour in that case is? Log and retry on next interval?

File: lib/utils/cageAttest.js [hunk 2]
```diff
@@ -33,13 +35,13 @@ function parseCageNameFromHost(hostname) {
   const hostnameTokens = hostname.split('.');
   // Check if nonce prefix is present
   if (hostnameTokens[1] === 'attest') {
-    return hostnameTokens[2];
+    return { cageName: hostnameTokens[2], appUuid: hostnameTokens[3] };
```
Comment (lfarrel6) [inline lib/utils/cageAttest.js:38]: The function name should be updated if it's returning this object

**Review by lfarrel6 — CHANGES_REQUESTED** [review]
Nice! Just 2 changes

File: lib/core/cageAttestationDoc.js (hunk 1 above)
Comment (hanneary) [reply lib/core/cageAttestationDoc.js]: I changed it to log on failure and reload doc for a cage on cache miss, the next interval would take too long

**Review by lfarrel6 — APPROVED** [review]

**Pull request merged**